projects/lights_control/src/lights_gpio.c

```diff
@@ -0,0 +1,212 @@
+  [LIGHTS_HORN_PERIPHERAL] = { .port = GPIO_PORT_B, .pin = 11 },  //
+};
+
+static const GPIOAddress s_high_beams_peripheral_addresses[NUM_LIGHTS_HIGH_BEAMS_PERIPHERALS] = {
+  [LIGHTS_HIGH_BEAMS_RIGHT_PERIPHERAL] = { .port = GPIO_PORT_B, .pin = 1 },  //
+  [LIGHTS_HIGH_BEAMS_LEFT_PERIPHERAL] = { .port = GPIO_PORT_B, .pin = 15 },  //
+};
+
+static const GPIOAddress s_low_beams_peripheral_addresses[NUM_LIGHTS_LOW_BEAMS_PERIPHERALS] = {
+  [LIGHTS_LOW_BEAMS_RIGHT_PERIPHERAL] = { .port = GPIO_PORT_B, .pin = 2 },  //
+  [LIGHTS_LOW_BEAMS_LEFT_PERIPHERAL] = { .port = GPIO_PORT_A, .pin = 8 },   //
+};
+
+static const GPIOAddress s_drl_peripheral_addresses[NUM_LIGHTS_DRL_PERIPHERALS] = {
+  [LIGHTS_DRL_RIGHT_PERIPHERAL] = { .port = GPIO_PORT_B, .pin = 0 },  //
+  [LIGHTS_DRL_LEFT_PERIPHERAL] = { .port = GPIO_PORT_A, .pin = 10 },  //
+};
+
+static const GPIOAddress s_brakes_peripheral_addresses[NUM_LIGHTS_BRAKES_PERIPHERALS] = {
+  [LIGHTS_RIGHT_REAR_BRAKE_PERIPHERAL] = { .port = GPIO_PORT_B, .pin = 1 },        //
+  [LIGHTS_RIGHT_REAR_OUTER_BRAKE_PERIPHERAL] = { .port = GPIO_PORT_B, .pin = 2 },  //
+  [LIGHTS_LEFT_REAR_BRAKE_PERIPHERAL] = { .port = GPIO_PORT_B, .pin = 15 },        //
+  [LIGHTS_LEFT_REAR_OUTER_BRAKE_PERIPHERAL] = { .port = GPIO_PORT_A, .pin = 8 },   //
+  [LIGHTS_CENTRE_BRAKE_PERIPHERAL] = { .port = GPIO_PORT_B, .pin = 14 },           //
+};
+
+static const GPIOAddress s_strobe_peripheral_addresses[NUM_LIGHTS_STROBE_PERIPHERALS] = {
+  [LIGHTS_STROBE_PERIPHERAL] = { .port = GPIO_PORT_B, .pin = 11 },  //
+};
+
+static const GPIOAddress
+    s_front_signal_left_peripheral_addresses[NUM_LIGHTS_FRONT_SIGNAL_LEFT_PERIPHERALS] = {
+      [LIGHTS_SIDE_LEFT_INDICATOR_PERIPHERAL] = { .port = GPIO_PORT_B, .pin = 14 },  //
+      [LIGHTS_LEFT_TURN_PERIPHERAL] = { .port = GPIO_PORT_A, .pin = 9 },             //
+    };
+
+static const GPIOAddress
+    s_front_signal_right_peripheral_addresses[NUM_LIGHTS_FRONT_SIGNAL_RIGHT_PERIPHERALS] = {
+      [LIGHTS_SIDE_RIGHT_INDICATOR_PERIPHERAL] = { .port = GPIO_PORT_B, .pin = 12 },  //
+      [LIGHTS_RIGHT_TURN_PERIPHERAL] = { .port = GPIO_PORT_B, .pin = 10 },            //
+    };
+
+static const GPIOAddress
+    s_rear_signal_left_peripheral_addresses[NUM_LIGHTS_REAR_SIGNAL_LEFT_PERIPHERALS] = {
+      [LIGHTS_LEFT_REAR_OUTER_TURN_PERIPHERAL] = { .port = GPIO_PORT_A, .pin = 10 },  //
+      [LIGHTS_LEFT_REAR_TURN_PERIPHERAL] = { .port = GPIO_PORT_A, .pin = 9 },         //
+    };
+
+static const GPIOAddress
+    s_rear_signal_right_peripheral_addresses[NUM_LIGHTS_REAR_SIGNAL_RIGHT_PERIPHERALS] = {
+      [LIGHTS_RIGHT_REAR_OUTER_TURN_PERIPHERAL] = { .port = GPIO_PORT_B, .pin = 0 },  //
+      [LIGHTS_RIGHT_REAR_TURN_PERIPHERAL] = { .port = GPIO_PORT_B, .pin = 10 },       //
+    };
+
+static const GPIOAddress s_board_type_address = { .port = GPIO_PORT_B, .pin = 13 };
+
+static LightsBoard s_board;
+
+static const GPIOAddress *s_address_front_lookup[] = {
```

Comment:
Keep in mind that since this an array, you'll end up with the maximum number of entries. Right now, both arrays will have 8 entries.